docs/under-the-hood/storage.md

```diff
@@ -1,138 +1,201 @@
-snapshot file provides a quicker way to restore the database state.
+According to the formula, storage memory usage should be: 
 
-Database recovery is done on startup from the most recent snapshot file. Since
-the snapshot may be older than the most recent update logged in the WAL file,
-the recovery process will apply the remaining state changes found in the WAL
-file.
+$
+\begin{aligned}
+\texttt{StorageRAMUsage} &= 21,723 \times 260\text{B} + 682,943 \times 180\text{B} \\ &= 5,647,980\text{B} + 122,929,740\text{B}\\ &= 128,577,720\text{B} \approx 125\text{MB}
+\end{aligned}
+$
 
-Behavior of these mechanisms can be tweaked in the configuration file,
-usually found in `/etc/memgraph/memgraph.conf`.
+Now, let's run an empty Memgraph instance on a x86 Ubuntu. It consumes **~75MB**
+of RAM due to baseline runtime overhead. Once the dataset is loaded, RAM usage
+rises up to **~260MB**. Memory usage primarily consists of storage and query
+execution memory usage. After executing `FREE MEMORY` query to force the cleanup
+of query execution, the RAM usage drops to **~200MB**. If the baseline runtime
+overhead of **75MB** is subtracted from the total memory usage of the dataset,
+which is **200MB**, and storage memory usage comes up to **~125MB**, which shows
+that the formula is correct.
 
-Check the reference guide on Configuration to see the [possible configuration
-settings for storage](../reference-guide/configuration#storage).
+## The calculation in detail
 
-:::caution
-Snapshot and WAL files are (currently) not compatible between Memgraph
-versions.
+Let's dive deeper into the memory usage values. Because Memgraph works on the
+x86 architecture, calculations are based on the x86 Linux memory usage.
+
+:::tip 
+For the latest and most precise memory layout please clone
+[Memgraph](https://github.com/memgraph/memgraph) and use, e.g.,
+[pahole](https://github.com/PhilArmstrong/pahole-gdb) to discover accurate
+information. 
 :::
 
-## Storable data types
+Each `Vertex` and `Edge` object has a pointer to a `Delta` object. The
+`Delta` object stores all changes on a certain `Vertex` or `Edge` and that's
+why `Vertex` and `Edge` memory usage will be increased by the memory of
+the `Delta` objects they are pointing to. If there are few updates, there are
+also few `Delta` objects because the latest data is stored in the object.
+But, if the database has a lot of concurrent operations, many `Delta` objects
+will be created. Of course, the `Delta` objects will be kept in memory as long as
+needed, and a bit more, because of the internal GC inefficiencies.
+
+### Delta memory layout
+
+Each `Delta` object has a least **104B**.
+
+### Vertex memory layout
+
+Each `Vertex` object has at least **112B** + **104B** for the `Delta` object, in
+total, a minimum of **216B**.
+
+### Edge memory layout
+
+Each `Edge` object has at least **40B** + **104B** for the `Delta` object, in
+total, a minimum of **144B**.
+
+### `SkipList` memory layout
```

Comment:
The issue here is that the term skiplist doesn't exist if it's not written as a code object... it's a skip list.. so I wanted to avoid confusion that SkipList is indeed SkipList... if I were to write this without code style, as a regular word, it should be skip list.. 
So whatever works for majority, but skiplist is not in anyway an ok word :)

Reply:
we can code all the objects

Reply:
aaa now I get it, feel free to choose. 😀

Reply:
I would leave the code 😄

Reply:
coded all the objects... have issues with this

> ### Index memory layout
> 
> Each `LabelIndex::Entry` object has exactly **16B**.
> 
> Depending on the actual value stored, each `LabelPropertyIndex::Entry` has at least **72B**.
> 
> Each Index Entry Object is placed into the `SkipList`.

What's the Index Entry Object... how should I capitalize that? Is that also in code as other objects? What object is it?